Whitelisting scala.Function1 crashes the compiler.

Whitelisting `scala/Function1.scala` will crash the compiler in `sbt test` with:

```scala
[error] Test dotc.tests.compileStdLib failed: java.lang.AssertionError: assertion failed: duplicate class Function1#14914; previous was class Function1#14919, took 1.176 sec
[error]     at scala.Predef$.assert(Predef.scala:165)
[error]     at dotty.tools.dotc.core.Scopes$MutableScope.enter(Scopes.scala:224)
[error]     at dotty.tools.dotc.core.SymDenotations$ClassDenotation.enterNoReplace(SymDenotations.scala:1542)
[error]     at dotty.tools.dotc.core.SymDenotations$ClassDenotation.enter(SymDenotations.scala:1524)
[error]     at dotty.tools.dotc.typer.NamerContextOps$class.enter(Namer.scala:35)
[error]     at dotty.tools.dotc.core.Contexts$Context.enter(Contexts.scala:57)
[error]     at dotty.tools.dotc.typer.Namer.enterSymbol(Namer.scala:335)
[error]     at dotty.tools.dotc.typer.Namer.dotty$tools$dotc$typer$Namer$$recur$1(Namer.scala:428)
[error]     at dotty.tools.dotc.typer.Namer.indexExpanded(Namer.scala:438)
[error]     at dotty.tools.dotc.typer.Namer$$anonfun$6.apply(Namer.scala:533)
[error]     at dotty.tools.dotc.typer.Namer$$anonfun$6.apply(Namer.scala:533)
[error]     at scala.collection.LinearSeqOptimized$class.foldLeft(LinearSeqOptimized.scala:124)
[error]     at scala.collection.immutable.List.foldLeft(List.scala:84)
[error]     at scala.collection.TraversableOnce$class.$div$colon(TraversableOnce.scala:136)
[error]     at scala.collection.AbstractTraversable.$div$colon(Traversable.scala:104)
[error]     at dotty.tools.dotc.typer.Namer.index(Namer.scala:533)
[error]     at dotty.tools.dotc.typer.Namer.dotty$tools$dotc$typer$Namer$$recur$1(Namer.scala:421)
[error]     at dotty.tools.dotc.typer.Namer.indexExpanded(Namer.scala:438)
[error]     at dotty.tools.dotc.typer.Namer.index(Namer.scala:411)
[error]     at dotty.tools.dotc.typer.FrontEnd$$anonfun$enterSyms$1.apply$mcV$sp(FrontEnd.scala:43)
[error]     at dotty.tools.dotc.typer.FrontEnd.monitor(FrontEnd.scala:23)
[error]     at dotty.tools.dotc.typer.FrontEnd.enterSyms(FrontEnd.scala:41)
[error]     at dotty.tools.dotc.typer.FrontEnd$$anonfun$runOn$2.apply(FrontEnd.scala:78)
[error]     at dotty.tools.dotc.typer.FrontEnd$$anonfun$runOn$2.apply(FrontEnd.scala:78)
[error]     at scala.collection.immutable.List.foreach(List.scala:381)
[error]     at dotty.tools.dotc.typer.FrontEnd.runOn(FrontEnd.scala:78)
[error]     at dotty.tools.dotc.Run$$anonfun$compileUnits$1$$anonfun$apply$mcV$sp$1.apply(Run.scala:75)
[error]     at dotty.tools.dotc.Run$$anonfun$compileUnits$1$$anonfun$apply$mcV$sp$1.apply(Run.scala:72)
[error]     at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
[error]     at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
[error]     at dotty.tools.dotc.Run$$anonfun$compileUnits$1.apply$mcV$sp(Run.scala:72)
[error]     at dotty.tools.dotc.Run$$anonfun$compileUnits$1.apply(Run.scala:66)
[error]     at dotty.tools.dotc.Run$$anonfun$compileUnits$1.apply(Run.scala:66)
[error]     at dotty.tools.dotc.util.Stats$.monitorHeartBeat(Stats.scala:76)
[error]     at dotty.tools.dotc.Run.compileUnits(Run.scala:66)
[error]     at dotty.tools.dotc.Run.compileSources(Run.scala:63)
[error]     at dotty.tools.dotc.Run.compile(Run.scala:47)
[error]     at dotty.tools.dotc.Driver.doCompile(Driver.scala:26)
[error]     at dotty.tools.dotc.Driver.process(Driver.scala:124)
[error]     at dotty.tools.dotc.Driver.process(Driver.scala:93)
[error]     at dotty.tools.dotc.CompilerTest.compileArgs(CompilerTest.scala:282)
[error]     at dotty.tools.dotc.CompilerTest.compileList(CompilerTest.scala:227)
[error]     at dotc.tests.compileStdLib(tests.scala:244)
```

The same happens for all 22 function types.

--- a/compiler/src/dotty/tools/dotc/core/Definitions.scala
+++ b/compiler/src/dotty/tools/dotc/core/Definitions.scala
@@ -834,7 +834,7 @@ class Definitions {
     val newDecls = new MutableScope(oldDecls) {
       override def lookupEntry(name: Name)(implicit ctx: Context): ScopeEntry = {
         val res = super.lookupEntry(name)
-        if (res == null && name.functionArity > 0)
+        if (res == null && name.functionArity > MaxImplementedFunctionArity)
           newScopeEntry(newFunctionNTrait(name.functionArity))
         else res
       }

